### ISSUE TITLE
Issues with shooting

### ISSUE THREAD

**Issue by PonasKovas** [feather-rs/feather#303]
Hello
in the readme you include "Arrow shooting" in the features list, so I expected them to work at least somewhat well, but this is what happens when you shoot:
![shooting](https://user-images.githubusercontent.com/25486263/90439482-4e586980-e0de-11ea-991f-ab2ec5d477f4.gif)

I know bouncing arrows are cool (very cool actually, I have never thought of that before), but I'm sure we don't want it.

Overall, the physics of arrows are pretty weird...

Another issue with the shooting system is that you can shoot in survival with no arrows (and it feels really weird since the client doesn't show the animation).

Also, in vanilla minecraft you have to pull the bow at least for a bit (0.2 second or something) before you can actually shoot, seems that this feature doesn't exist in feather, since I can just right click the bow and spam the arrows as fast as I want to.

I'd love to contribute, if you think that these issues (or at least some of them) are not too complicated for a newbie in this project, it'd be awesome if you could point me in the right direction and I would try to fix them.

**Comment by caelunshun** [comment]
Yep, arrow shooting is still preliminary. Physics need to be reworked in many areas as well.

The problems here:
* The physics implementation was tested primarily with item entities, which behave much differently than arrows (they generally have lower velocity, and they don't stick to surfaces). A major fix here would be to allow some entities to "stick" to blocks on impact, such that they are unaffected by gravity and further velocity changes. Internally, this would be implemented by adding a `SurfaceStick` (or similar) marker component to arrow entities, and then having the physics implementation test whether this component is present.
* As for the pulling delay, that should be pretty easy to implement—all the arrow shooting code is in `server/player/src/packet_handlers/digging.rs`. There's an `ItemTimedUse` component which stores the number of ticks the bow's been pulled back for. We can insert a check that returns early if that number is fewer than 4.
* It looks like the code for consuming arrows in survival was already implemented, but I commented it out because at the time survival mode was unimplemented. I've pushed a fix for this.
